packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_place_auction_bid.test.ts

```diff
@@ -0,0 +1,893 @@
+      expect(prevouts.length).toStrictEqual(1)
+      expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+      expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+
+      await alice.generate(1)
+      await tGroup.waitForSync()
+
+      const aliceColAccAfter = await alice.rpc.account.getAccount(aliceColAddr)
+      expect(aliceColAccAfter).toStrictEqual(['300000.00000000@DFI', '29999.00000000@BTC', '9400.00000000@TSLA'])
+
+      const auctions = await alice.container.call('listauctions')
+      expect(auctions[0]).toStrictEqual({
+        vaultId: bobVaultId,
+        liquidationHeight: 168,
+        batchCount: 2,
+        liquidationPenalty: 5,
+        loanSchemeId: 'scheme',
+        ownerAddress: bobVaultAddr,
+        state: 'inLiquidation',
+        batches: [{
+          index: 0,
+          collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+          loan: '500.00399539@TSLA',
+          highestBid: {
+            amount: '526.00000000@TSLA',
+            owner: bobColAddr
+          }
+        },
+        {
+          index: 1,
+          collaterals: ['5000.00000000@DFI', '0.50000000@BTC'],
+          loan: '500.00399539@TSLA',
+          highestBid: {
+            amount: '600.00000000@TSLA',
+            owner: aliceColAddr
+          }
+        }]
+      })
+    }
+
+    // end the auction and alice win the bid
+    await bob.generate(36)
+
+    const auctions = await alice.container.call('listauctions')
+    expect(auctions).toStrictEqual([]) // no more auctions
+
+    const bobColAccEndBid = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobColAccEndBid).toStrictEqual(['13900.00000000@DFI', '0.50000000@BTC', '19.45454546@TSLA'])
+
+    const aliceColAccEndBid = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceColAccEndBid).toStrictEqual(['305000.00000000@DFI', '29999.50000000@BTC', '9400.00000000@TSLA'])
+
+    const vault = await alice.container.call('getvault', [bobVaultId])
+    expect(vault.state).toStrictEqual('active')
+    expect(vault.collateralAmounts).toStrictEqual([])
+    expect(vault.loanAmounts).toStrictEqual([])
+    expect(vault.interestAmounts).toStrictEqual([])
+    expect(vault.collateralValue).toStrictEqual(0)
+    expect(vault.loanValue).toStrictEqual(0)
+    expect(vault.interestValue).toStrictEqual(0)
+    expect(vault.collateralRatio).toStrictEqual(-1)
+    expect(vault.informativeRatio).toStrictEqual(-1)
+  })
+
+  it('test super bid recover vault state to active against other bids on other batches', async () => {
+    const bobAccBefore = await bob.rpc.account.getAccount(bobColAddr)
+    expect(bobAccBefore).toStrictEqual(['8900.00000000@DFI', '545.45454546@TSLA'])
+
+    const aliceAccBefore = await alice.rpc.account.getAccount(aliceColAddr)
+    expect(aliceAccBefore).toStrictEqual(['300000.00000000@DFI', '29999.00000000@BTC', '10000.00000000@TSLA'])
+
+    {
+      const txn = await bBuilder.loans.placeAuctionBid({
```

Comment:
If I understand the test purpose correctly, this bid should be removed. The batch index 0 should end without bidder and go back to active vault as normal collateral and loan token. Assert vault has collateral and loan after auction end

Reply:
i misunderstood your pre-request.. lemme refine it